docs/nlu/choosing-a-pipeline.rst

```diff
@@ -15,76 +13,133 @@ it on your dataset.
-for your domain. For example, in general English, the word "balance" is closely related to "symmetry",
-but very different to the word "cash". In a banking domain, "balance" and "cash" are closely related
-and you'd like your model to capture that. This pipeline doesn't use a language-specific model,
-so it will work with any language that you can tokenize (on whitespace or using a custom tokenizer).
+A pipeline usually consist of three main parts:
 
-You can read more about this topic `here <https://medium.com/rasa-blog/supervised-word-vectors-from-scratch-in-rasa-nlu-6daf794efcd8>`__ .
+    1. Tokenizaion
+    2. Featuirzation
+    3. Entity Recognition / Intent Classification / Response Selectors
 
-MITIE
-~~~~~
+Tokenization
+~~~~~~~~~~~~
+If your chosen language is whitespace-tokenized (words are separated by spaces), you
+can use the ``WhitespaceTokenizer``. If this is not the case you should use a different tokenizer.
+We support a number of different :ref:`tokenizers <tokenizers>`, or you can :ref:`create your own <custom-nlu-components>`.
 
-You can also use MITIE as a source of word vectors in your pipeline, see :ref:`section_mitie_pipeline`. The MITIE backend performs well for small datasets, but training can take very long if you have more than a couple of hundred examples.
+.. note::
+    Some components further down the pipeline may require a specific tokenizer. You can find those requirements
+    on the individual components in :ref:`components`. If a required component is missing inside the pipeline, an
+    error will be thrown.
+
+Featurization
+~~~~~~~~~~~~~
+You need to decide whether to use components that provide pre-trained word embeddings or not.
+
+If you do not use any pre-trained word embeddings, your word vectors will be customised for your domain. For example,
+in general English, the word "balance" is closely related to "symmetry", but very different to the word "cash". In a
+banking domain, "balance" and "cash" are closely related and you'd like your model to capture that. If you don't
+use any pre-trained word embeddings inside your pipeline, you are not bound to a specific language and domain.
+Thus, you should only use featurizers from the category `sparse` featuirzers, such as
+``CountVectorsFeaturizer`` or ``RegexFeaturizer``.
+
+The advantage of using pre-trained word embeddings in your pipeline is that if you have a training example like:
+"I want to buy apples", and Rasa is asked to predict the intent for "get pears", your model already knows that the
+words "apples" and "pears" are very similar. This is especially useful if you don't have large enough training data.
+We support a few components that provide pre-trained word embeddings:
+
+1. ``MitieFeaturizer``
+2. ``SpacyFeaturizer``
+3. ``ConveRTFeaturizer``
+4. ``LanguageModelFeaturizer``
+
+The advantage of the ``ConveRTFeaturizer`` is that it doesn't treat each word of the user message independently, but
```

Comment:
add link to polyai

Reply:
Added link to polyai in the docs for actual component. Added a `:ref:` here.